### ISSUE TITLE
M2-3: Data Validation Methodology

### ISSUE THREAD

**Issue by bouthilx** [bouthilx/compute-forecast-framework#46]
# M2-3: Data Validation Methodology

## Objective
Extend quality framework for extraction validation with confidence scoring and cross-validation.

**Worker**: Analysis | **Effort**: S(2-3h) | **Domain**: Infrastructure

### Extraction Quality Validator
```python
from dataclasses import dataclass
from typing import Dict, List, Optional, Any, Tuple
from enum import Enum
from src.quality.quality_analyzer import QualityAnalyzer
from src.quality.metrics import QualityMetrics

class ExtractionQuality(Enum):
    HIGH = "high"           # >90% confidence
    MEDIUM = "medium"       # 70-90% confidence
    LOW = "low"            # 50-70% confidence
    UNRELIABLE = "unreliable"  # <50% confidence

@dataclass
class ExtractionValidation:
    paper_id: str
    extraction_type: str  # gpu_hours, memory, parameters, etc.
    extracted_value: Any
    confidence: float
    quality: ExtractionQuality
    validation_method: str
    cross_validation_result: Optional[Dict[str, Any]] = None

class ExtractionQualityValidator(QualityAnalyzer):
    """Extends quality framework for extraction validation"""
    
    def __init__(self):
        super().__init__()
        self.completeness_weights = {
            "gpu_hours": 0.3,
            "gpu_type": 0.2,
            "training_time": 0.2,
            "parameters": 0.15,
            "dataset_size": 0.15
        }
    
    def validate_extraction(self, 
                          paper: Paper, 
                          extraction: ComputationalAnalysis) -> ExtractionValidation:
        """Validate single extraction with confidence scoring"""
        pass
    
    def calculate_completeness_score(self, extraction: Dict[str, Any]) -> float:
        """Score extraction completeness (0-1)"""
        pass
    
    def cross_validate_extractions(self, 
                                 manual: Dict[str, Any], 
                                 automated: Dict[str, Any]) -> Dict[str, Any]:
        """Compare manual vs automated extraction"""
        # Returns:
        # {
        #   "agreement_score": 0.85,
        #   "discrepancies": [...],
        #   "confidence": 0.9
        # }
        pass
```

### Consistency Checker
```python
@dataclass
class ConsistencyCheck:
    check_type: str  # temporal, cross_paper, domain_specific
    passed: bool
    confidence: float
    details: Dict[str, Any]

class ExtractionConsistencyChecker:
    """Check consistency across similar papers"""
    
    def __init__(self):
        self.similarity_threshold = 0.8
        self.outlier_z_score = 3.0
    
    def check_temporal_consistency(self, 
                                  papers: List[Paper], 
                                  metric: str) -> ConsistencyCheck:
        """Verify metrics follow temporal trends"""
        # GPU hours should generally increase over time
        # Model sizes should follow known scaling laws
        pass
    
    def check_cross_paper_consistency(self, 
                                    paper_group: List[Paper], 
                                    extraction_field: str) -> ConsistencyCheck:
        """Check if similar papers have similar metrics"""
        pass
    
    def identify_outliers(self, 
                         values: List[float], 
                         context: Dict[str, Any]) -> List[int]:
        """Statistical outlier detection"""
        pass
```

### Cross-Validation Framework
```python
class CrossValidationFramework:
    """Framework for manual vs automated validation"""
    
    def __init__(self):
        self.agreement_threshold = 0.8
        self.manual_sample_size = 100
    
    def select_validation_sample(self, 
                               papers: List[Paper], 
                               stratify_by: str = "year") -> List[Paper]:
        """Select representative sample for manual validation"""
        pass
    
    def compare_extractions(self, 
                          manual: Dict[str, Any], 
                          automated: Dict[str, Any]) -> Dict[str, float]:
        """Compare extraction results"""
        # Returns accuracy metrics per field:
        # {
        #   "gpu_hours": 0.92,
        #   "parameters": 0.88,
        #   "training_time": 0.85
        # }
        pass
    
    def generate_calibration_model(self, 
                                 comparisons: List[Dict[str, Any]]) -> Any:
        """Build model to calibrate automated extractions"""
        pass
```

### Outlier Detection and Verification
```python
from scipy import stats
import numpy as np

class OutlierDetector:
    """Statistical outlier detection for extracted values"""
    
    def __init__(self):
        self.methods = ["z_score", "iqr", "isolation_forest"]
    
    def detect_outliers(self, 
                       values: List[float], 
                       method: str = "z_score") -> List[int]:
        """Detect outlier indices"""
        pass
    
    def contextualize_outlier(self, 
                            paper: Paper, 
                            field: str, 
                            value: float) -> Dict[str, Any]:
        """Provide context for why value might be outlier"""
        # Check if:
        # - Paper describes novel architecture
        # - Different domain than majority
        # - Breakthrough paper with citations
        pass
    
    def verify_outlier(self, 
                      paper: Paper, 
                      extraction: Dict[str, Any]) -> bool:
        """Manual verification prompt for outliers"""
        pass
```

### Integration with Existing Quality Framework
```python
from src.quality.adaptive_thresholds import AdaptiveThresholdCalculator
from src.quality.metrics import QualityMetrics

class IntegratedExtractionValidator:
    def __init__(self):
        # Use existing components
        self.threshold_calculator = AdaptiveThresholdCalculator()
        self.quality_metrics = QualityMetrics()
        
        # Add extraction-specific validators
        self.extraction_validator = ExtractionQualityValidator()
        self.consistency_checker = ExtractionConsistencyChecker()
    
    def validate_extraction_batch(self, 
                                extractions: List[ComputationalAnalysis]) -> Dict[str, Any]:
        """Comprehensive validation using all systems"""
        pass
```

### Completeness Rules
```python
completeness_rules = {
    "gpu_required": ["gpu_hours", "gpu_type"],
    "training_required": ["training_time", "parameters"],
    "optional": ["batch_size", "learning_rate", "optimizer"]
}
```

### Consistency Rules
```python
consistency_rules = {
    "gpu_hours_vs_parameters": "exponential_relationship",
    "year_vs_model_size": "increasing_trend",
    "domain_specific": {
        "nlp": {"parameters": [1e6, 1e12]},
        "cv": {"gpu_hours": [10, 10000]}
    }
}
```

## Success Criteria
- [ ] Extraction completeness scoring operational
- [ ] Cross-validation framework comparing manual/automated
- [ ] Consistency checks preventing obvious errors
- [ ] Outlier detection calibrated (not too sensitive)
- [ ] Integration with existing quality framework
- [ ] Confidence scores correlate with actual accuracy
- [ ] 90%+ agreement between manual and automated

## File Structure
```
src/quality/extraction/
├── extraction_validator.py
├── consistency_checker.py
├── cross_validation.py
├── outlier_detection.py
└── validation_rules.yaml
```

## Implementation Timeline
- Hour 1: Extraction validator and completeness scoring
- Hour 2: Consistency checking and outlier detection
- Hour 3: Cross-validation framework and integration
